en/advanced-topics/memory-leak-detector.md

```diff
@@ -0,0 +1,54 @@
+# Native engine memory leak detection system
+
+The native engine is developed using the C++ language, and memory leaks are inevitable. In order to facilitate game & engine developers to quickly find memory leaks, Cocos Creator v3.4.0 provides a **memory leak detection system**
+
+Compared with other memory leak detection tools, what are the advantages of the built-in memory leak detection tool in Cocos Creator:
+- **Cross-platform**: support Windows/Android/Mac/iOS platforms.
+- **Ease of use**: No need to download additional tools and perform complex configurations; support output of stack information at memory leaks, which is convenient for quickly locating leaks.
+- **Consistency**: The usage process of each platform is almost the same: start the game from the IDE -> run for a period of time -> close the game -> view the IDE output log.
+- **Real-time**: Although the frame rate of the game in the profiling mode has dropped, it still maintains the real-time running frame rate.
+- **Accuracy**: theoretically zero false positives.
+
+Steps for usage
+1. The memory leak detection system is turned off by default. To enable it, you need to modify the value of the macro `USE_MEMORY_LEAK_DETECTOR` in `engine-native/cocos/base/Config.h` to `1`.
+    ```c++
+    #ifndef USE_MEMORY_LEAK_DETECTOR
+        #define USE_MEMORY_LEAK_DETECTOR 1
+    #endif
+    ```
+2. Due to the different implementation mechanisms, two additional things need to be done on the android platform:
+    - Add a line as follows in the `project directory/native/engine/android/CMakeLists.txt`:
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -finstrument-functions")
+        ```
+        set(PROJ_SOURCES
+            ${CMAKE_CURRENT_LIST_DIR}/../common/Classes/Game.h
+            ${CMAKE_CURRENT_LIST_DIR}/../common/Classes/Game.cpp
+            ${CMAKE_CURRENT_LIST_DIR}/jni/main.cpp
+        )
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -finstrument-functions")
+        ```
+    - If the Job System is TBB when building the android project, please change to None or TaskFlow (default), because TBB will conflict with the current memory capture implementation mechanism on the android platform:
```

Comment:
```suggestion

    - If the Job System is TBB when building the android project, please change to None or TaskFlow (default), because TBB will conflict with the current memory capture implementation mechanism on the Android platform:
```